File: src/main/java/no/nav/service/pensjon/regler/to/FaktoromregnInntektBatchRequest.java
package no.nav.service.pensjon.regler.to;

import java.util.Date;
import java.util.Vector;

import no.nav.domain.pensjon.regler.gomregning.KravFaktoromregningGrunnlag;

public class FaktoromregnInntektBatchRequest extends ServiceRequest {


	/**
	 * Dato for når gammel regulering gjaldt.
	 */
	private Date gammelReguleringGjaldtDato;

	/**
	 * Ny dato det skal omregnes for.
	 */
	private Date nyReguleringOmregnFraDato;

	/**
	 * Liste av grunnlag for faktoromregning.
	 */
	private Vector<KravFaktoromregningGrunnlag> kravFaktoromregningGrunnlagListe = new Vector<KravFaktoromregningGrunnlag>();

	/**
	 * Angir om tjenestekallet er en batch jobb eller ikke.
	 */
	private boolean automaticJob;
}

File: src/main/kotlin/no/nav/pensjon/regler/to/FaktoromregnInntektBatchRequest.kt
package no.nav.pensjon.regler.to

import no.nav.pensjon.regler.domain.gomregning.KravFaktoromregningGrunnlag
import java.util.*
import java.time.LocalDate

class FaktoromregnInntektBatchRequest : ServiceRequest() {
    /**
     * Dato for når gammel regulering gjaldt.
     */
    var gammelReguleringGjaldtDatoLd: LocalDate? = null

    /**
     * Ny dato det skal omregnes for.
     */
    var nyReguleringOmregnFraDatoLd: LocalDate? = null

    /**
     * Liste av grunnlag for faktoromregning.
     */
    var kravFaktoromregningGrunnlagListe = Vector<KravFaktoromregningGrunnlag>()

    /**
     * Angir om tjenestekallet er en batch jobb eller ikke.
     */
    var automaticJob = false
}